clients/src/test/java/org/apache/kafka/clients/producer/internals/BufferPoolTest.java

```diff
@@ -175,9 +175,9 @@ public void testBlockTimeout() throws Exception {
         } catch (TimeoutException e) {
             // this is good
         }
-        assertTrue("available memory" + pool.availableMemory(), pool.availableMemory() >= 9 && pool.availableMemory() <= 10);
```

Comment:
Instead of removing this, we should probably just say that it can be from 8 to 10 with a comment saying that thread scheduling sometimes means that no deallocation happens by the time we get here.